src/ui/components/filters/filterboxcomponent.js

```diff
@@ -40,13 +40,13 @@ class FilterBoxConfig {
      * If true, show a "reset all" button to reset all facets
      * @type {boolean}
      */
-    this.resetFilters = config.resetFacets === undefined ? true : config.resetFacets;
+    this.resetFilters = config.resetFilters === undefined ? !config.searchOnChange : config.resetFilters;
 
     /**
      * The label to show for the "reset all" button
      * @type {string}
      */
-    this.resetFiltersLabel = config.resetFacetsLabel || 'reset all';
+    this.resetFiltersLabel = config.resetFiltersLabel || 'reset all';
```

Comment:
Was the Facets component also setting resetFacetsLabel and resetFiltersLabel to be the same?

Reply:
yup! what it's currently doing is passing resetFiltersLabel: config.resetFacetsLabel, as well as passing the whole facets config in the object assign https://github.com/yext/answers/blob/master/src/ui/components/filters/facetscomponent.js#L203